scripts/lib/install-node

```diff
@@ -1,9 +1,9 @@
 #!/usr/bin/env bash
-set -e
+set -uo pipefail
 
 ZULIP_PATH="$(dirname "$0")/../.."
 ZULIP_SRV="/srv"
-if [ "$TRAVIS" ] ; then
+if [[ -n "${TRAVIS:-}" ]] ; then
```

Comment:
I'd prefer to keep this as  -- we don't use the `[[` style in any of our shell scripts, and I don't see the advantage of migrating.  Though this is an improvement:
`if [ -n "$TRAVIS" ]; ...`

Reply:
`"${TRAVIS:-}"` (or just `"${TRAVIS-}"`) is for appeasing the `set -u` added above, which is probably a good idea overall.

(Agreed that `[[` isn’t worth changing to. No particular opinion on `-n`—both `[ "str" ]` and `[ -n "str" ]` are correct and common, although `[ -n "str" ]` does seems more common for us.)

Reply:
Just so there's some background, my suggestions come from a few years of writing shell scripts, using shellcheck and from the bash guide I linked above. In this case you can find some information [here](http://mywiki.wooledge.org/BashGuide/TestsAndConditionals#Conditional_Blocks_.28if.2C_test_and_.5B.5B.29). A quick quote:

> Good Practice:
> Whenever you're making a Bash script, you should always use [[ rather than [.

If the scripts are working fine there's no reason to migrate on `[[`, just keep in mind it's the preferred way of writing bash scripts today making it more readable and familiar for others reading your codebase.